sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultAccessControlClientBuilder.java

```diff
@@ -129,13 +135,25 @@ public KeyVaultAccessControlAsyncClient buildAsyncClient() {
 
         String clientName = properties.getOrDefault(SDK_NAME, "UnknownName");
         String clientVersion = properties.getOrDefault(SDK_VERSION, "UnknownVersion");
+        String applicationId =
+            clientOptions == null ? httpLogOptions.getApplicationId() : clientOptions.getApplicationId();
 
-        policies.add(new UserAgentPolicy(httpLogOptions.getApplicationId(), clientName, clientVersion,
-            buildConfiguration));
+        policies.add(new UserAgentPolicy(applicationId, clientName, clientVersion, buildConfiguration));
         HttpPolicyProviders.addBeforeRetryPolicies(policies);
         policies.add(retryPolicy == null ? new RetryPolicy() : retryPolicy);
         this.policies.add(new KeyVaultCredentialPolicy(credential));
         policies.addAll(this.policies);
+
+        if (clientOptions != null) {
+            Iterable<Header> headers = clientOptions.getHeaders();
+
+            if (headers.iterator().hasNext()) {
+                List<HttpHeader> httpHeaderList = new ArrayList<>();
+                headers.forEach(header -> httpHeaderList.add(new HttpHeader(header.getName(), header.getValue())));
+                policies.add(new AddHeadersPolicy(new HttpHeaders(httpHeaderList)));
+            }
```

Comment:
This can be simplified.

```suggestion
           
                List<HttpHeader> httpHeaderList = new ArrayList<>();
                clientOptions.getHeaders().forEach(header -> httpHeaderList.add(new HttpHeader(header.getName(), header.getValue())));
                policies.add(new AddHeadersPolicy(new HttpHeaders(httpHeaderList)));
           
```

Reply:
@srnagar I think we should check for nullness for `clientOptions`.